File: apps/python-lua/rewardFunction.lua
local reward = 0.0
if passedCheckpoint then
    reward = (reward + 1.0)
end
if isCarReversed then
    reward = (reward - 1.0)
else
    reward = (reward + (distanceToCheckpoint * 0.03))
    reward = (reward + (speed * 0.02))
end
if (speed > 0.0) then
    reward = (reward + 0.05)
end
local count = 0
for index, value in ipairs(distanceToWall) do
    if ((count == 0) or (count == 2)) then
        if (value < 0.25) then
            reward = (reward - 1.0)
        end
    end
    if ((count == 1) or (count == 5)) then
        if (value < 0.1) then
            reward = (reward - 1.0)
        end
    end
    count = (count + 1)
    ::loop_label_1::
end
if isOutBound then
    reward = (reward - 1.0)
end
return reward

File: release/trainer/settings - Copy/rewardFunction.py
reward = 0.0

if passedCheckpoint:
    reward += 1.0

reward += (distanceToCheckpoint * 0.03)
reward += (speed * 0.02)

count = 0
for index, value in ipairs(distanceToWall):
    if (count == 0) or (count == 2) or (count == 3) or (count == 4) or (count == 6):
        if value < 0.25:
            reward -= 1.0
    if (count == 1) or (count == 5):
        if value < 0.1:
            reward -= 1.0
    count += 1
    
if isOutBound:
    reward -= 1.0

return reward
